File: src/main/java/com/elchologamer/userlogin/util/BaseListener.java
package com.elchologamer.userlogin.util;

import com.elchologamer.userlogin.UserLogin;
import org.bukkit.event.Listener;

public abstract class BaseListener implements Listener {

    private final UserLogin plugin = UserLogin.getPlugin();

    public void register() {
        plugin.getServer().getPluginManager().registerEvents(this, plugin);
    }

    public UserLogin getPlugin() {
        return plugin;
    }
}

File: src/main/kotlin/com/elchologamer/userlogin/listener/BaseListener.kt
package com.elchologamer.userlogin.listener

import com.elchologamer.userlogin.UserLogin.Companion.plugin
import org.bukkit.event.Listener

abstract class BaseListener : Listener {
    fun register() {
        plugin.server.pluginManager.registerEvents(this, plugin)
    }
}